### PR TITLE
Fix #190 | Some inflected forms not showing up

### LINKED ISSUE #190: Inflected forms of POS-ambiguous lemmas are not recognized: ê-pipohk, ê-kîsikâk
In conjunction with #185, some inflected forms corresponding to lemmas that have two POS are not recognized nor linked to the lemma reading with the right POS. For example:

- [ ] ê-pipok -> pipon (VII)
- [ ] ê-kîsikâk -> kîsikâw (VII)

### PR DESCRIPTION
**Pull request by Madoshakalaka**
The cause for #190 is quite trivial, `list` was used instead of `set` where unwanted duplication can happen, which in turn made the importer thinks the lemmas of the entries of _pipon_ in the xml is ambiguous.

The _ê-kîsikâk_ case might be caused by the same problem. The desired lemma shows up when I test locally.

I've built a full production database with the fix, and I'll upload it to the server once this pr is merged.

### CODE REVIEW

File: CreeDictionary/DatabaseManager/xml_entry_lemma_finder.py [hunk 1]
```diff
@@ -29,7 +29,9 @@ def extract_fst_lemmas(
     logger = DatabaseManagerLogger(__name__, verbose)
     logger.info("Determining lemma analysis for (xml_lemma, xml_pos, xml_lc) tuples...")
 
-    xml_lemma_pos_lc_to_analysis = dict()  # type: Dict[Tuple[str, str, str], ConcatAnalysis]
+    xml_lemma_pos_lc_to_analysis = (
+        dict()
+    )  # type: Dict[Tuple[str, str, str], ConcatAnalysis]
```
Comment (Madoshakalaka) [inline CreeDictionary/DatabaseManager/xml_entry_lemma_finder.py:34]: My IDE did some formatting automatically with black.

File: CreeDictionary/tests/DatabaseManager_tests/test_xml_importer_regressions.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    kill_him_inflections = []
+    for inflection in query_set:
+        for definition in inflection.definitions.all():
+            if "Kill" in definition.text:
+                kill_him_inflections.append(inflection)
+
+    assert len(kill_him_inflections) == 1
+    kill_him_inflection = kill_him_inflections[0]
+    assert kill_him_inflection.pos == "V"
```
Comment (Madoshakalaka) [inline CreeDictionary/tests/DatabaseManager_tests/test_xml_importer_regressions.py:21]: Old regression code, I changed the file name from `test_xml_importer_regression_0.py` to `test_xml_importer_regressions.py` to just host more tests in one file

File: CreeDictionary/tests/DatabaseManager_tests/test_xml_importer_regressions.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    # The Cree word pipon has two entries in the test xml, one's word class is VII and the other's is NI
+    migrate_and_import(shared_datadir / "crkeng-pipon-of-different-word-classes")
+
+    # todo: let `migrate_and_import` report success/ambiguity/no-analysis count so that further tests to the importer
+    #   can be easier constructed. e.g. in this case we'll only need to assert `success == 2`
+
+    assert (
+        Wordform.objects.filter(text="pipon", is_lemma=True, as_is=False).count() == 2
+    )
```
Comment (Madoshakalaka) [inline CreeDictionary/tests/DatabaseManager_tests/test_xml_importer_regressions.py:39]: new regression test

File: CreeDictionary/DatabaseManager/xml_entry_lemma_finder.py (hunk 1 above)
Comment (eddieantonio) [reply CreeDictionary/DatabaseManager/xml_entry_lemma_finder.py:34]: Haha, black HATES same line comments. Which brings up a good point: why _isn't_ this a regular type annotation?

File: CreeDictionary/DatabaseManager/xml_entry_lemma_finder.py [hunk 4]
```diff
@@ -74,7 +78,9 @@ def extract_fst_lemmas(
 
             if xml_lemma in xml_lemma_to_pos_lc:
                 for pos, lc in xml_lemma_to_pos_lc[xml_lemma]:
-                    xml_lemma_pos_lc_to_analysis[(xml_lemma, pos, lc)] = ConcatAnalysis("")
+                    xml_lemma_pos_lc_to_analysis[(xml_lemma, pos, lc)] = ConcatAnalysis(
+                        ""
```
Comment (eddieantonio) [inline CreeDictionary/DatabaseManager/xml_entry_lemma_finder.py:82]: I think Black is trying to tell us something... 😂

File: CreeDictionary/DatabaseManager/xml_entry_lemma_finder.py [hunk 5]
```diff
@@ -109,16 +115,16 @@ def extract_fst_lemmas(
                 xml_lemma
             ]:  # for each pos, lc determine which is the analysis
 
-                ambiguities: List[ConcatAnalysis] = []
+                ambiguities: Set[ConcatAnalysis] = set()
```
Comment (eddieantonio) [inline CreeDictionary/DatabaseManager/xml_entry_lemma_finder.py:118]: Yeah, it helps if we deduplicate ambiguities!

File: CreeDictionary/tests/DatabaseManager_tests/test_xml_importer_regressions.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    # https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/issues/190
+    # Issue description: search results for some inflected form of word pipon is not showing up
+    # Cause: pipon lemmas wrongly marked as "as-is" in the database when the xml actually provided enough resolution
+    # on the word classes (VII and NI)
+
+    # The Cree word pipon has two entries in the test xml, one's word class is VII and the other's is NI
+    migrate_and_import(shared_datadir / "crkeng-pipon-of-different-word-classes")
+
+    # todo: let `migrate_and_import` report success/ambiguity/no-analysis count so that further tests to the importer
+    #   can be easier constructed. e.g. in this case we'll only need to assert `success == 2`
```
Comment (Madoshakalaka) [inline CreeDictionary/tests/DatabaseManager_tests/test_xml_importer_regressions.py:35]: I proposed this here and I want to do it later in a different PR. Our importer already has beautify output to stdout:

```
xx entries have (lemma, pos, lc) duplicate to others. Their definition will be merged
xx (xml_lemma, pos, lc) tuples extracted
xx (lemma, pos, lc) found proper lemma analysis
There are xx (lemma, pos, lc) that the fst can not give any analyses.
There are xx (lemma, pos, lc) that do not have proper lemma analysis by fst
There are xx (lemma, pos, lc) that have ambiguous lemma analyses
```
When I was debugging the issue, stdout told me 
"**0** Found proper analysis. There are **2** that have ambiguous lemma analyses"
 which made me realize the issue immediately. 
And I knew I fixed the issue as it turned:
"**2** Found proper analysis. There are **0** that have ambiguous lemma analyses"

So returning the digits will help in building tests.

File: CreeDictionary/tests/DatabaseManager_tests/data/crkeng-pipon-of-different-word-classes/crkeng.xml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+       <tg xml:lang="eng">
+           <t pos="V" sources="MD">It is winter.</t>
+       </tg>
+       </mg>
+       <mg>
+       <tg xml:lang="eng">
+           <t pos="V" sources="CW">it is winter; is one year</t>
+       </tg>
+       </mg>
+    </e>
```
Comment (Madoshakalaka) [inline CreeDictionary/tests/DatabaseManager_tests/data/crkeng-pipon-of-different-word-classes/crkeng.xml:39]: minimal producing input for the regression test.

File: CreeDictionary/tests/DatabaseManager_tests/test_xml_importer_regressions.py (hunk 6 above)
Comment (eddieantonio) [inline CreeDictionary/tests/DatabaseManager_tests/test_xml_importer_regressions.py:35]: 👍

File: cypress/integration/regressions.spec.js [hunk 8]
```diff
@@ -214,4 +214,19 @@ context('Regressions', () => {
...
+   */
+  it('should not present un-related translation for preverbs', function () {
+    cy.visitSearch('ê-kîsikâk')
+
+    // there should be only one result
+    cy.get('[data-cy=search-results]')
+      .should('contain', 'kîsikâw')
+  })
+
+
```
Comment (Madoshakalaka) [inline cypress/integration/regressions.spec.js:231]: regression for the other case.

File: CreeDictionary/DatabaseManager/xml_entry_lemma_finder.py (hunk 4 above)
Comment (Madoshakalaka) [reply CreeDictionary/DatabaseManager/xml_entry_lemma_finder.py:82]: black is a paranoid :b:

**Pull request merged**